staging/src/k8s.io/apimachinery/pkg/runtime/mapper.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package runtime
+
+import (
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type resourceMapper struct {
+	// keyFunc computes a key for the specified resource (this allows honoring colocated resources across API groups).
+	// if null, or if "" is returned, resource.String() is used as the key
+	keyFunc func(resource schema.GroupResource) string
+	// resources maps key -> subresource -> equivalent resources (subresource is not included in the returned resources)
+	resources map[string]map[string][]schema.GroupVersionResource
+	// kinds maps resource -> subresource -> kind
+	kinds map[schema.GroupVersionResource]map[string]schema.GroupVersionKind
+	// keys caches the computed key for each GroupResource
+	keys map[schema.GroupResource]string
+}
+
+var _ ResourceMapper = (*resourceMapper)(nil)
+
+// NewResourceMapper creates a resource mapper that considers all versions of a GroupResource to be equivalent.
+func NewResourceMapper() ResourceMapper {
+	return &resourceMapper{}
+}
+
+// NewResourceMapperWithKeyFunc creates a resource mapper with a custom key function.
+// The provided function is called to determine the key for each group resource.
```

Comment:
so it is actually an identity func.

Reply:
changed name to identity func and updated doc